api/diagnostics.go

```diff
@@ -298,118 +258,125 @@ func (j *DiagnosticsJob) flushReport(zipWriter *zip.Writer, fileName string, rep
 	}
 }
 
-func (j *DiagnosticsJob) collectDataFromNodes(ctx context.Context, nodes []dcos.Node, summaryReport *bytes.Buffer,
-	summaryErrorsReport *bytes.Buffer) ([]string, error) {
-
-	fetchRequests := j.getEndpointsToFetch(ctx, nodes, summaryReport, summaryErrorsReport)
-
-	fetchReq := make(chan fetcher.EndpointRequest, len(fetchRequests))
-	for _, r := range fetchRequests {
-		fetchReq <- r
-	}
-	close(fetchReq)
+func (j *DiagnosticsJob) downloadFromNodes(ctx context.Context, bundleID string,
+	nodes []dcos.Node, zipWriter *zip.Writer, summaryReport *bytes.Buffer, summaryErrorsReport *bytes.Buffer) {
 
-	fetchStatusUpdate := make(chan fetcher.StatusUpdate)
-	fetchResponse := make(chan fetcher.BulkResponse)
+	requests := j.getBundleRequests(bundleID, nodes, summaryReport)
+	requestFinished := make(chan fetcher.RequestStatus)
 
-	numberOfWorkers := j.Cfg.FlagDiagnosticsBundleFetchersCount
-	for i := 0; i < numberOfWorkers; i++ {
-		f, err := fetcher.New(j.Cfg.FlagDiagnosticsBundleDir, j.client, fetchReq, fetchStatusUpdate, fetchResponse, j.FetchPrometheusVector)
+	// send creation requests to all nodes
+	for _, r := range requests {
+		err := r.SendCreationRequest(ctx, requestFinished)
```

Comment:
we should probably make this asynchronous in the same way as the `downloadBundle` call is asynchronous.

Also, `SendCreationRequest` should always send sth. to `requestFinished` and not return anything so that the number of items in the channel is equal to the number of requests.